pkg/controllers/provisioning/scheduling/topology.go

```diff
@@ -268,7 +269,7 @@ func (t *Topology) countDomains(ctx context.Context, tg *TopologyGroup) error {
 		}
 		// nodes may or may not be considered for counting purposes for topology spread constraints depending on if they
 		// are selected by the pod's node selectors and required node affinities.  If these are unset, the node always counts.
-		if !tg.nodeFilter.Matches(node) {
+		if tg.nodeAffinityPolicy == &honor && !tg.nodeFilter.Matches(node) {
```

Comment:
You've already edited the filter here so do you need to apply the affinity policy again here? I think at this point you can just check if the nodeFilter matches since the filter won't filter anything out due to your `NewTopologyNodeFilter` change